javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javaparsermodel/contexts/CompilationUnitContext.java

```diff
@@ -152,19 +145,19 @@ public CompilationUnitContext(CompilationUnit wrappedNode, TypeSolver typeSolver
             // ("A.B.C.D"), we look for the outermost type ("A" in the previous example) first, then recursively invoke
             // this method for the remaining part of the given name.
             if (name.indexOf('.') > -1) {
+
                 SymbolReference<ResolvedTypeDeclaration> ref = null;
-                SymbolReference<ResolvedTypeDeclaration> outerMostRef =
-                    solveType(name.substring(0, name.indexOf(".")));
-                if (outerMostRef != null && outerMostRef.isSolved() &&
-                    outerMostRef.getCorrespondingDeclaration() instanceof JavaParserClassDeclaration) {
-                    ref = ((JavaParserClassDeclaration) outerMostRef.getCorrespondingDeclaration())
-                        .solveType(name.substring(name.indexOf(".") + 1));
-                } else if (outerMostRef != null && outerMostRef.isSolved() &&
-                    outerMostRef.getCorrespondingDeclaration() instanceof JavaParserInterfaceDeclaration) {
-                    ref = ((JavaParserInterfaceDeclaration) outerMostRef.getCorrespondingDeclaration())
-                        .solveType(name.substring(name.indexOf(".") + 1));
+                Optional<? extends ResolvedTypeDeclaration> outerMostRef = solveType(name.substring(0, name.indexOf(".")))
```

Comment:
here this variable is a(n optional) type declaration, rather than reference -- please rename the variable to reflect this